src/controllers/controller.bar.js

```diff
@@ -159,21 +166,27 @@ module.exports = function(Chart) {
 		},
 
 		calculateBarWidth: function(ruler) {
-			var xScale = this.getScaleForId(this.getMeta().xAxisID);
+			var me = this;
+			var meta = me.getMeta();
+			var xScale = me.getScaleForId(meta.xAxisID);
 			if (xScale.options.barThickness) {
 				return xScale.options.barThickness;
 			}
-			return xScale.options.stacked ? ruler.categoryWidth : ruler.barWidth;
+			return (xScale.options.stacked && meta.stack === undefined) ? ruler.categoryWidth : ruler.barWidth;
```

Comment:
Done and changed unit tests. 

I also changed the controller method names - `getBarCount` to `getStackCount` and `getBarIndex` to `getStackIndex`. The ruler property `barCount` is now `stackCount`. Related unit tests changed too.